src/main/java/io/quarkus/search/app/util/UrlInputProvider.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.search.app.util;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import io.quarkus.logging.Log;
+import io.quarkus.search.app.hibernate.InputProvider;
+
+public class UrlInputProvider implements InputProvider {
+
+    private static final byte[] NO_CONTENT_CONTENT = "<!DOCTYPE html><html lang=\"en\"><head><meta charset=\"UTF-8\"><title>Quarkus</title></head><body><!--No Content--></body></html>"
+            .getBytes(StandardCharsets.UTF_8);
+
+    private final Path temporaryFile;
+    private final URL url;
+
+    public UrlInputProvider(String uri) {
+        try {
+            this.url = new URI(uri).toURL();
+            Path temporaryFile = Files.createTempDirectory("quarkiverse-")
+                    .resolve(this.url.getPath().replaceAll("[./]", "-"));
+            try (InputStream inputStream = this.url.openStream()) {
+                Files.copy(inputStream, temporaryFile);
```

Comment:
Could you try to avoid the copying when the URI scheme is "file://"?

Also, when is this file deleted?
We'll probably want to use a common directory for all temp files of a given indexing and delete that directory after indexing... See `CloseableDirectory#temp`.

Reply:
>Could you try to avoid the copying when the URI scheme is "file://"?

This shouldn't happen as the `UrlInputProvider` should only be created for those quarkiverse guides that "by definition" are pointing to some remote HTML page. so I didn't do anything about it. But I've used the closable directory so the files will be removed after indexing.